### PR TITLE
[Windows] Notify changes in Picker ItemsSource

### LINKED ISSUE #9239: ItemsSource cannot be changed on a Picker on Windows
### Description

On a `Picker`, it's not possible to change the `ItemsSource` after it has been set on Windows (works find on Android). At least not easily. Every time you try to change, the previous set list is used instead.

### Steps to Reproduce

1. Create a new MAUI application.
2. Replace MainPage.xaml:

``` XAML
    <VerticalStackLayout 
        Spacing="25" 
        Padding="30,0" 
        VerticalOptions="Start">

        <HorizontalStackLayout Spacing="25">
            <VerticalStackLayout>
                <Label Text="Available colors:" />
                <Label Text="{Binding Colors[0]}" />
                <Label Text="{Binding Colors[1]}" />
                <Label Text="{Binding Colors[2]}" />
            </VerticalStackLayout>

            <VerticalStackLayout>
                <Label Text="Select a color:" />
                <Picker ItemsSource="{Binding Colors}" SelectedIndex="{Binding SelectedColorIndex}" />
            </VerticalStackLayout>
        </HorizontalStackLayout>

        <Button 
            x:Name="CounterBtn"
            Text="Show blue colors"
            Clicked="OnCounterClicked"
            HorizontalOptions="Start"
            />
        
    </VerticalStackLayout>
```

3. Replace MainPage.xaml.cs:

``` C#
public MainPage()
{
    InitializeComponent();

    BindingContext = this;
}

public List<string> Colors { get; set; } = new List<string>()
{
    "Red",
    "LightRed",
    "DarkRed",
};

public int SelectedColorIndex { get; set; }

private int ChangeCount = 0;

private void OnCounterClicked(object sender, EventArgs e)
{
    ChangeCount++;
    Colors = new List<string>()
    {
        "Blue " + ChangeCount,
        "LightBlue " + ChangeCount,
        "DarkBlue " + ChangeCount,
    };

    OnPropertyChanged(nameof(Colors));
    SelectedColorIndex = 0;
    OnPropertyChanged(nameof(SelectedColorIndex));
}
```

4. Run the application. The default red colors are visible:

image

5. Press the button to change to blue colors. They are not changed in the picker:

image

7. Press the button again. Now they are changed in the picker, but the previous list is used:

image

Full repro: https://github.com/pekspro/MauiIssues/tree/9239_ItemsSource_cannot_be_changed_on_a_Picker

### Version with bug

6.0.408 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10.0.17763.0

### Did you find any workaround?

Using an `ObservableCollection<string>` works if it is set once and the individual items are changed.

### Relevant log output

_No response_

### LINKED ISSUE #9496: .NET MAUI Picker ItemsSource binding failed
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/NET-MAUI-Picker-ItemsSource-binding-fai/10116249)._

---
[severity:It's more difficult to complete my work] [regression] [worked-in:17.3.0 Preview 5]
After update Picker in .NET MAUI stopped binding ItemsSource. I tried with List and ObservableCollection. Before update binding was working fine.

---
#### Feedback Bot on 8/10/2022, 11:05 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Notify changes in Picker ItemsSource (without using ObservableCollection). Fix the issue #9239 updating the Picker ItemsSource.

fix-9239

### Issues Fixed

Fixes #9239 
Fixes https://github.com/dotnet/maui/issues/9496

### CODE REVIEW

File: src/Controls/src/Core/Picker.cs [hunk 1]
```diff
@@ -318,6 +318,7 @@ void ResetItems()
 			((LockableObservableListWrapper)Items).InternalClear();
 			foreach (object item in ItemsSource)
 				((LockableObservableListWrapper)Items).InternalAdd(GetDisplayMember(item));
+			Handler?.UpdateValue(nameof(IPicker.Items));
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Picker.cs:321]: This is the main change.

**Review by rmarinho — APPROVED** [review]

File: src/Core/tests/DeviceTests/Stubs/PickerStub.cs [hunk 2]
```diff
@@ -12,7 +13,24 @@ public partial class PickerStub : StubBase, IPicker
...
+			get => _itemsSource;
+			set => SetProperty(ref _itemsSource, value, onChanged: OnItemsSourceChanged);
+		}
+
+		void OnItemsSourceChanged(IList oldValue, IList newValue)
+		{
+			if (ItemsSource == null)
+				return;
+
+			Items.Clear();
```
Comment (mohachouch) [inline src/Core/tests/DeviceTests/Stubs/PickerStub.cs:29]: I think you have to clear the list before check ItemsSource is null

Comment (rmarinho) [reply src/Core/tests/DeviceTests/Stubs/PickerStub.cs:29]: Look @mohachouch  our WPF guru :D 

This is just a test so i _think_ is fine. 

For when a WPF backend to MAUI @mohachouch  ? i can help :P

**Comment by tekmun** [comment]
I downloaded the entire dotnet MAUI zip repo. I run .\build.ps1 and my .nuget and .dotnet directories in my user folder are updated. But after I clean and rebuild, the issue still remains. I suspect I am not using this latest build in my project.
Can someone guide me how I can specify this latest dotnet MAUI in my project instead of using the Nuget version? What must I change in my .csproj file? Is there a documentation? I want to test that the latest fix really solve this issue in my project.
Thanks.

**Comment by acrigney** [comment]
Yes this is still a problem for me. The Items is set when my view model is created but the binding doesn't work.

**Pull request merged**